core/src/main/java/org/infinispan/commands/read/AbstractDataCommand.java

```diff
@@ -14,22 +16,33 @@
  */
 public abstract class AbstractDataCommand extends AbstractTopologyAffectedCommand implements DataCommand {
    protected Object key;
+   private long flags;
```

Comment:
Hey @danberindei @pruivo what kind of optimization this is? This field hides `flags` in `AbstractFlagAffectedCommand` and breaks `printFlags()`. And more importantly, it increases the size of the object.